packages/devtools_app/lib/src/debugger/evaluate.dart

```diff
@@ -24,17 +29,142 @@ class ExpressionEvalField extends StatefulWidget {
   _ExpressionEvalFieldState createState() => _ExpressionEvalFieldState();
 }
 
-class _ExpressionEvalFieldState extends State<ExpressionEvalField> {
-  TextEditingController textController;
-  FocusNode textFocus;
+class _AutoCompleteController extends DisposableController
+    with SearchControllerMixin, AutoCompleteSearchControllerMixin {}
+
+class _ExpressionEvalFieldState extends State<ExpressionEvalField>
+    with SearchFieldMixin, AutoDisposeMixin {
+  _AutoCompleteController _autoCompleteController;
   int historyPosition = -1;
 
+  final evalTextFieldKey = GlobalKey(debugLabel: 'evalTextFieldKey');
+
   @override
   void initState() {
     super.initState();
 
-    textController = TextEditingController();
-    textFocus = FocusNode();
+    _autoCompleteController = _AutoCompleteController();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    addAutoDisposeListener(_autoCompleteController.searchNotifier, () {
```

Comment:
you need to cancel the existing listeners at the start of this method to avoid ending up with n unbounded number of listeners.
What dependency has changed? I'm not spotting why the autoCompleteController needs a new listener here instead of in initState.

Reply:
Oversight. Moved to initState.